src/fn/watch-project.js

```diff
@@ -0,0 +1,177 @@
+const environment = require('../lib/environment');
+const process = require('process');
+const path = require('path');
+const { error, info } = require('../lib/log');
+const Queue = require('queue-promise');
+const watcher = require('@parcel/watcher');
+const convertForms = require('../lib/convert-forms');
+const uploadForms = require('../lib/upload-forms');
+const convertContactForm = require('./convert-contact-forms').execute;
+const { uploadAppSettings } = require('./upload-app-settings');
+const compileAppSettings = require('./compile-app-settings').execute;
+const uploadCustomTranslations = require('./upload-custom-translations').execute;
+const uploadResources = require('./upload-resources').execute;
+
+const formXLSRegex = /^[a-zA-Z0-9_-]+\.xlsx$/;
+const formPropertiesRegex = /^[a-zA-Z0-9_-]+\.properties.json$/;
+const formMediaRegex = /^[a-zA-Z0-9_]+(?:-media)$/;
+const formXMLRegex = /^[a-zA-Z0-9_-]+\.xml$/;
+
+const uploadInitialState = (api) => {
+    return Promise.all(
+        [
+            uploadResources(),
+            uploadForms(environment.pathToProject, 'app'),
+            uploadForms(environment.pathToProject, 'contact', { id_prefix: 'contact:', default_context: { person: false, place: false } }),
+            uploadCustomTranslations(),
+            uploadAppSettings(api)
+        ]
+    );
+};
+
+let fsEventsSubscription;
+let eventQueue;
+
+const cleanUp = async () => {
+    if (fsEventsSubscription) await fsEventsSubscription.unsubscribe();
+    if(eventQueue) await eventQueue.clear();
+};
```

Comment:
For safety, please also unset `fsEventsSubscription` and `eventQueue` here. 
```suggestion
    if (fsEventsSubscription) {
        await fsEventsSubscription.unsubscribe();
        fsEventsSubscription = undefined;
    }
    if(eventQueue) {
        await eventQueue.clear();
        eventQueue = undefined; 
    }
```